pkg/storage/stores/shipper/bloomshipper/cache.go

```diff
@@ -133,6 +130,12 @@ func (b BlockDirectory) Release() error {
 	return nil
 }
 
+func (b BlockDirectory) Size() int64 {
+	bloomFileStats, _ := os.Lstat(path.Join(b.Path, v1.BloomFileName))
```

Comment:
IIUC, the size won't change since we are not modifying the block. Maybe we can do the Lstat on `NewBlockDirectory` save the size inside the  `BlockDirectory` struct? That way we can save a bunch of sys-calls.